### PR TITLE
Update genesis dockerfile build for github sources.

### PR DESCRIPTION
**Pull request by abizjak**
Use the public repositories and public docker images.

### CODE REVIEW

File: scripts/genesis-tools.Dockerfile [hunk 1]
```diff
@@ -10,14 +10,14 @@
...
 
 # Which branch of concordium-base to build the tools from.
-ARG BASE_REF=master
+ARG BASE_REF=main
 
-RUN mkdir -p -m 0600 ~/.ssh && ssh-keyscan gitlab.com >> ~/.ssh/known_hosts
+RUN mkdir -p -m 0600 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
 
-RUN --mount=type=ssh git clone --depth 1 --branch ${BASE_REF} git@gitlab.com:Concordium/concordium-base.git
+RUN --mount=type=ssh git clone --depth 1 --branch ${BASE_REF} git@github.com:Concordium/concordium-base.git
```
Comment (bisgardo) [inline scripts/genesis-tools.Dockerfile:20]: `git clone --depth 1 --branch main https://github.com/Concordium/concordium-base.git` would be sufficient.

Comment (bisgardo) [reply scripts/genesis-tools.Dockerfile:20]: (can't make multiline suggestions)

Comment (abizjak) [reply scripts/genesis-tools.Dockerfile:20]: I can change. 

But the reason I did not is that there might be a case where we'll need a private fork for security fixes, in which case we'll need this again.

Are you strongly against keeping it like this?

Comment (bisgardo) [reply scripts/genesis-tools.Dockerfile:20]: Not strongly against it, no.

Comment (abizjak) [reply scripts/genesis-tools.Dockerfile:20]: Ok, then I'll keep it if only for myself to not forget how to do this in the future.

**Pull request merged**